packages/astro/src/core/endpoint/index.ts

```diff
@@ -31,19 +31,26 @@ type EndpointCallResult =
 			response: Response;
 	  };
 
+type CreateAPIContext = {
+	request: Request;
+	params: Params;
+	site?: string;
+	props: Record<string, any>;
+	adapterName?: string;
+};
```

Comment:
The feature sounds good to me. I have a small concern if this param for `createAPIContext` is too wide though. Other than `request`, is it easily possible for integrations to derive `params`, `site`, `props` and `adapterName`?

`site` and `adapterName` could be something Astro automatically handles.

Reply:
@matthewp and I discussed it. We landed to these conclusions:
- `adapterName` is the name of the adapter itself (integration)
- `props`, `site` and `params` can be empty. `params` be derived from the `request`, but an adapter might have a different type from Astro `Params` - which is a `Record<string, string | undefined>`. 


Should we expose a small function that wraps the original `createAPIContext` with narrowed types? Ideally, only request, params and adapterName. What do you think? I like the idea

Reply:
Yeah I think it would be great to narrow the types. Could we also exclude the `adapterName`? The value (like `site`) can already be found from the config in build-time.